### PR TITLE
Added script for requirement [PolicyTableUpdate] Support of http flow…

### PR DESCRIPTION
**Pull request by maria-mih**
… of Policy Table Update
TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua 
To run the script you need to replace
user_modules/shared_testcases/testCasesForPolicyTable.lua
user_modules/shared_testcases/testCasesForBuildingSDLPolicyFlag.lua

from #32. Common functions are under review.
@istoimenova Please review

### CODE REVIEW

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---------------------------------------------------------------------------------------------
+-- Requirements summary: 
+--     [PolicyTableUpdate] Support of "http" flow of Policy Table Update
+--
+-- Description: 
+-- SDL should be successfully built "-DEXTENDED_POLICY:HTTP" flag is empty
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih Should be EXTENDED_POLICY:empty

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+-- 1. Performed steps
+-- Build SDL 
+--
+-- Expected result:
+-- SDL is successfully built
+-- The flag -DEXTENDED_POLICY:HTTP has no value
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih Should be EXTENDED_POLICY: has no value.

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+-- PTU passes successfully 
+
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih remove empty line

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+--[[ Required Shared libraries ]]
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+--local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonPreconditions = require ('user_modules/shared_testcases/commonPreconditions')
+local testCasesForPolicyTable = require('user_modules/shared_testcases/testCasesForPolicyTable')
+local testCasesForBuildingSDLPolicyFlag = require('user_modules/shared_testcases/testCasesForBuildingSDLPolicyFlag')
+  
+--[[ General Precondition before ATF start ]]
+-- commonFunctions:SDLForceStop()
+testCasesForBuildingSDLPolicyFlag:Update_PolicyFlag("ENABLE_HTTP_POLICY")
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih  The call of function should be:
testCasesForBuildingSDLPolicyFlag:Update_PolicyFlag("EXTENDED_POLICY","")

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+testCasesForBuildingSDLPolicyFlag:CheckPolicyFlagAfterBuild("ENABLE_HTTP_POLICY")
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih The call of function should be
testCasesForBuildingSDLPolicyFlag:CheckPolicyFlagAfterBuild("EXTENDED_POLICY","")

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+commonSteps:DeleteLogsFileAndPolicyTable()
+commonPreconditions:Connecttest_without_ExitBySDLDisconnect_WithoutOpenConnectionRegisterApp("connecttest_RAI.lua")
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih This is redundant as you are using "require('connecttest')

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  config.defaultProtocolVersion = 2
+  
+--[[ General Settings for configuration ]]
+  Test = require('connecttest')
+  require('cardinalities')
+  require('user_modules/AppTypes')
+  --local mobile_session = require('mobile_session')
+ 
+--[[ Test ]]
+ --ToDo: Function should be debugged! Runtime error!
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih Remove the comment. The function is working with "require('user_modules/AppTypes')"

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+ --commonFunctions:newTestCasesGroup("TC_PTU_HTTP_FLOW")
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih Uncomment the line. The function is working with "require('user_modules/AppTypes')"

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+--[[ Test ]]
+ --ToDo: Function should be debugged! Runtime error!
+ --commonFunctions:newTestCasesGroup("TC_PTU_HTTP_FLOW") 
+  function Test:HTTP_Flow_AfterBuild ()
+   testCasesForPolicyTable:flow_PTU_SUCCEESS_EXTERNAL_HTTP (self)
+  end
+
+  
+--[[ Postconditions ]]
+ --ToDo: Function should be debugged! Runtime error!
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih Remove the comment. The function is working with "require('user_modules/AppTypes')"

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+-- commonFunctions:newTestCasesGroup("Postconditions")
+-- function Test:Postcondition_SDLForceStop()
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih Uncomment the function. The function is working with "require('user_modules/AppTypes')"

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+ --ToDo: Function should be debugged! Runtime error!
+ --commonFunctions:newTestCasesGroup("TC_PTU_HTTP_FLOW") 
+  function Test:HTTP_Flow_AfterBuild ()
+   testCasesForPolicyTable:flow_PTU_SUCCEESS_EXTERNAL_HTTP (self)
+  end
+
+  
+--[[ Postconditions ]]
+ --ToDo: Function should be debugged! Runtime error!
+-- commonFunctions:newTestCasesGroup("Postconditions")
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih Uncomment the line. The function is working with "require('user_modules/AppTypes')"

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+  Test = require('connecttest')
+  require('cardinalities')
+  require('user_modules/AppTypes')
+  --local mobile_session = require('mobile_session')
+ 
+--[[ Test ]]
+ --ToDo: Function should be debugged! Runtime error!
+ --commonFunctions:newTestCasesGroup("TC_PTU_HTTP_FLOW") 
+  function Test:HTTP_Flow_AfterBuild ()
+   testCasesForPolicyTable:flow_PTU_SUCCEESS_EXTERNAL_HTTP (self)
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih Please add comment that function is still not implemented. To be seen in script.

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+-- PTU passes successfully 
+
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+
+--[[ Required Shared libraries ]]
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+--local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih Uncomment the line.  See comments below for functions that are using commonFunctions.

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+commonPreconditions:Connecttest_without_ExitBySDLDisconnect_WithoutOpenConnectionRegisterApp("connecttest_RAI.lua")
+ 
+  --ToDo (mmihaylova-banska): Should be removed when issue: "ATF does not stop HB timers by closing session and connection" is fixed
+  config.defaultProtocolVersion = 2
+  
+--[[ General Settings for configuration ]]
+  Test = require('connecttest')
+  require('cardinalities')
+  require('user_modules/AppTypes')
+  --local mobile_session = require('mobile_session')
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih Remove the line as is not used.

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+ 
+--[[ Test ]]
+ --ToDo: Function should be debugged! Runtime error!
+ --commonFunctions:newTestCasesGroup("TC_PTU_HTTP_FLOW") 
+  function Test:HTTP_Flow_AfterBuild ()
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @maria-mih Rename the function as Test:TestStep_HTTP_Flow_AfterBuild() according to template: https://github.com/CustomSDL/sdl_atf_test_scripts/pull/32

File: test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua (hunk 1 above)
Comment (maria-mih) [inline test_scripts/Polices/build_options/TC002_SDL_Build_HTTP_Flow_DEXTENDED_POLICY_No_Flag.lua]: @istoimenova Done please check commit 090760d64ed0fa0

**Comment by maria-mih** [comment]
@istoimenova 
Please review, the function for success flow is under review in pull #407

**Pull request merged**